index.js

```diff
@@ -3,6 +3,26 @@
 var clone = require('101/clone')
 var debug = require('debug')('swarmerode')
 var exists = require('101/exists')
+var Promise = require('bluebird')
+var cache = {}
+
+function handleCache (key, cacheFetch, cb) {
+  // If SWARMERODE_CACHE_LENGTH is not set we don't want to cache anything
+  if (!exists(process.env.SWARMERODE_CACHE_LENGTH)) {
+    return cacheFetch(cb)
+  }
+
+  if (!cache[key]) {
+    cache[key] = Promise.fromCallback(cacheFetch)
+    cache[key]
```

Comment:
we want to wait for the cacheFetch to actually resolve before starting the timer.